stock_picking_auto_revert/models/stock_picking.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2019 ForgeFlow
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, exceptions, _
+from odoo.tools import float_compare
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.multi
+    def action_revert_recreate(self):
+        self.ensure_one()
+        pick = self
+        msg = _(
+            "This picking cannot be returned because the products "
+            "are not available in the destination location."
+        )
+        # Create return picking
+        StockReturnPicking = self.env["stock.return.picking"]
+        default_data = StockReturnPicking.with_context(
+            active_ids=pick.ids, active_id=pick.id
+        ).default_get(
+            [
+                "move_dest_exists",
+                "original_location_id",
+                "product_return_moves",
+                "parent_location_id",
+                "location_id",
+            ]
+        )
+        for rm in default_data["product_return_moves"]:
+            if len(rm) > 2 and isinstance(rm[2], dict):
+                sm = pick.move_lines.filtered(
+                    lambda x: x.product_id.id == rm[2]["product_id"]
+                )
```

Comment:
Hi, if "pick" has 2 lines having the same product, "sm" may not ensure having 1 record. so, "sm.product_uom_qty" (at the line 43) will return an error in that case.

Reply:
Good point. Corrected.